modules/tinymce/src/core/test/ts/webdriver/delete/ShortcutDeleteTest.ts

```diff
@@ -0,0 +1,116 @@
+import * as InlineFormatDelete from 'tinymce/core/delete/InlineFormatDelete';
+
+type BackspaceDeleteModifier = { ctrl: true } | { alt: true } | { meta: true };
+
+describe('webdriver.tinymce.core.delete.ShortcutDeleteTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [], true);
+  const platform = PlatformDetection.detect();
+  const os = platform.os;
+  const browser = platform.browser;
+
+  const doBackspaceDelete = async (editor: Editor, deletionKey: 'Backspace' | 'Delete', modifiers: BackspaceDeleteModifier): Promise<void> => {
+    await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo(modifiers, deletionKey) ]);
+    if (browser.isSafari()) {
+      // safari doesn't trigger this override automatically in test
+      InlineFormatDelete.refreshCaret(editor);
+    }
+  };
+
+  const assertStructureAndSelectionFormattedWord = (editor: Editor): void => {
+    TinyAssertions.assertContentStructure(editor,
+      ApproxStructure.build((s, str, _arr) => {
+        return s.element('body', {
+          children: [
+            s.element('p', {
+              // firefox preserves formats
+              children: browser.isFirefox()
+                ? [
+                  s.element('strong', {
+                    children: [
+                      s.element('em', {
+                        children: [
+                          s.element('span', {
+                            attrs: {
+                              style: str.is('text-decoration: underline;')
+                            },
+                            children: [
+                              s.element('span', {
+                                attrs: {
+                                  'id': str.is('_mce_caret'),
+                                  'data-mce-bogus': str.is('1')
+                                }
+                              })
+                            ]
+                          })
+                        ]
+                      })
+                    ]
+                  })
+                ]
+                : [
+                  s.element('span', {
+                    attrs: {
+                      'id': str.is('_mce_caret'),
+                      'data-mce-bogus': str.is('1')
+                    }
+                  })
+                ]
+            })
+          ]
+        });
+      })
+    );
+    const selPath = browser.isFirefox() ? [ 0, 0, 0, 0, 0, 0 ] : [ 0, 0, 0 ];
+    TinyAssertions.assertSelection(editor, selPath, 0, selPath, 0);
+  };
+
+  const assertContentDeletionThenTyping = (editor: Editor): void =>
+    TinyAssertions.assertContent(editor, browser.isFirefox()
+      ? '<p><span style="text-decoration: underline;">d</span></p>'
+      : '<p>d</p>');
+
+  const ctrlModifier: BackspaceDeleteModifier = os.isMacOS() ? { alt: true } : { ctrl: true };
+
+  it('Ctrl/Alt + Backspace at the end of a formatted word', async () => {
```

Comment:
Shouldn't these also have the ticket id prefix?

Reply:
I'm not sure what the best practice here is. The ticket is on some specific behaviours, which is why I added the ticket id prefixes for the tests that check that the browser no longer adds rogue tags after ranged deletion and then entering text.

The fix resulted in implementing a new override that has a larger scope than the ticket specified. I didn't mark these tests with the ticket ids as they are not directly related to the ticket requirements, but rather related to the fix.

Reply:
Ticket id prefixed all tests added.